examples/mnist-convnet.py

```diff
@@ -72,16 +86,23 @@ def _build_graph(self, input_vars):
         train_error = tf.reduce_mean(wrong, name='train_error')
         summary.add_moving_summary(train_error)
 
-        # Use a regex to find parameters to apply weight decay.
-        # Here we apply a weight decay on all W (weight matrix) of all fc layers
-        wd_cost = tf.mul(1e-5,
-                         regularize_cost('fc.*/W', tf.nn.l2_loss),
-                         name='regularize_loss')
-        summary.add_moving_summary(cost, wd_cost)
-
-        # monitor histogram of all weight (of conv and fc layers) in tensorboard
-        summary.add_param_summary([('.*/W', ['histogram'])])
-        self.cost = tf.add_n([wd_cost, cost], name='cost')
+        if use_tfslim == False:
+            # Use a regex to find parameters to apply weight decay.
+            # Here we apply a weight decay on all W (weight matrix) of all fc layers
+            wd_cost = tf.mul(1e-5,
+                             regularize_cost('fc.*/W', tf.nn.l2_loss),
+                             name='regularize_loss')
+            summary.add_moving_summary(cost, wd_cost)
+
+            # monitor histogram of all weight (of conv and fc layers) in tensorboard
+            summary.add_param_summary([('.*/W', ['histogram'])])
+            self.cost = tf.add_n([wd_cost, cost], name='cost')
+        else:
+            regul_loss = tf.get_collection(tf.GraphKeys.REGULARIZATION_LOSSES)
```

Comment:
I think it would be better to add this to any trainer,
In the model it can be turned off and on by

````python
with slim.arg_scope([slim.layers.conv2d, ...], normalizer_fn=slim.layers.batch_norm, weights_regularizer=slim.l2_regularizer(0.001),
````